### PR TITLE
Updated NvTransform to do scale, crop and color transform 

### LINKED ISSUE #175: Have Crop , Scale and Color Transform in Single NvTransform Module
**Is your feature request related to a problem? Please describe.**
single module for crop, scale, and color transform 
ability to select different nvtransform filters

### PR DESCRIPTION
**Pull request by yashrajsapra**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #175 
**Checklist**

- [x] I have read the [Contribution Guidelines](https://github.com/Apra-Labs/ApraPipes/wiki/Contribution-Guidelines)
- [x] I have written Unit Tests
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach

### CODE REVIEW

File: base/test/nvtransform_tests.cpp [hunk 1]
```diff
@@ -126,4 +127,48 @@ BOOST_AUTO_TEST_CASE(fdtest)
 	DMAUtils::freeCudaPtr(eglInImage,&pInResource, eglDisplay);
 }
 
+BOOST_AUTO_TEST_CASE(crop, *boost::unit_test::disabled())
```
Comment (kumaakh) [inline base/test/nvtransform_tests.cpp:130]: Please add similar tests with FileReader, if these tests remain "disabled" we will ahve regressions.

**Review by kumaakh — CHANGES_REQUESTED** [review]
need tests which run during build.

**Review by kumaakh — CHANGES_REQUESTED** [review]
need tests which show if feature works... there are other smaller comments as well

**Comment by yashrajsapra** [comment]
Yeah Akhil, I will work on that

**Comment by yashrajsapra** [comment]
> need tests which show if feature works... there are other smaller comments as well

Hi Akhil,
Need to push a new module for doing a memory conversion from host to dma

**Review by mraduldubey — APPROVED** [review]

**Pull request merged**